tm2/pkg/bft/consensus/replay.go

```diff
@@ -307,6 +307,13 @@ func (h *Handshaker) ReplayBlocks(
 			return nil, err
 		}
 
+		// Save the results by height
+		abciResponse := sm.NewABCIResponsesFromNum(len(res.TxResponses))
+		copy(abciResponse.DeliverTxs, res.TxResponses)
+		sm.SaveABCIResponses(h.stateDB, 0, abciResponse)
+
+		// NOTE: we don't save results by tx hash since the transactions are in the AppState opaque type
```

Comment:
Here is a crazy idea: what do you think about saving the genesis block (`block 0`) to the DB?

I know we don't have all of the information for it, but I think having the majority of it fetch-able will allow for better integration with existing endpoints (specifically `tx`), and not require us to do the dance of (`min`) that we do for RPC endpoints in this PR. It just makes every client's life easier without having to "know" that `0` is unfetchable for _some_ endpoints (currently all)

Reply:
I agree it would be better but I'm not sure how to do this without importing gno.land types into tm2

Reply:
There's also the whole block 0 vs block 1 ordeal

Reply:
For lurkers:

Block 1 is using the state derived in InitChain as base (it's a special case in BeginBlock) but is pulling txs from the mempool, so genesis transasction seem to belong more to some "Block 0"